netsuite/client.py

```diff
@@ -83,8 +84,59 @@ def wrapper(self, *args, **kw):
     return decorator
 
 
+class NetSuiteTransport(Transport):
+    """
+    NetSuite dynamic domain wrapper for zeep.transports.transport
+
+    Latest NetSuite WSDL now uses relative definition addresses
+
+    zeep maps reflective remote calls to the base WSDL address,
+    rather than the dynamic subscriber domain
+
+    Wrap the zeep transports service with our address modifications
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Assign the dynamic host domain component to a class variable
+        """
+
+        _wsdl = kwargs['wsdl_url']
+
+        kwargs.pop('wsdl_url',None)
+        
+        # Find first slash after domain name
+        idx = _wsdl.index('/', 8) + 1
+
+        # We only want the dynamic domain
+        _wsdl=_wsdl[:idx]
+        self._wsdl = _wsdl
```

Comment:
Another way to do this that might be a little less verbose and uses the standard URL parsing library:

```python
from urllib.parse import urlparse

class NetSuiteTransport(Transport):
    def __init__(self, wsdl_url, *args, **kwargs):
        parsed_wsdl_url = urlparse(wsdl_url)
        self._wsdl = f'{parsed_wsdl_url.scheme}://{parsed_wsdl_url.netloc}/'
        super().__init__(**kwargs)
```

Reply:
I think this is another issue with P3 / P2 compatibility.  The relevant module for P2 is urlparse, whereas in P3 it is urllib.parse.

Reply:
The original author was only coding and testing for Python 3. See https://github.com/merit-network/netsuite/pull/2#discussion_r388426592